Provider/src/main/java/dev/openfeature/contrib/providers/ConfidenceFeatureProvider.kt

```diff
@@ -54,14 +58,43 @@ class ConfidenceFeatureProvider private constructor(
             eventsPublisher.publish(OpenFeatureEvents.ProviderReady)
```

Comment:
Maybe we should to send `ProviderError` if the cache is empty at this point? Or `ProviderStale`?

Reply:
this exception happens when the network request has failed. especially when the initialisation strategy is activate and fetch IMO it doesn't make sense to have provider not ready. for fetch and activate, maybe .. but is it what we want? considering that if the developer decides to have the strategy as "try to refresh first" or "force refresh first"? in the latter case they can't use the properties since the provider is never ready. they will be stuck. but if the definition of FetchAndActivate is try to fetch, then the failure in the network request should be treated as ready? we have to add this to the documentation anyways.

Reply:
I agree, with ActivateAndFetch, the provider should be ready, since the developer would most likely apply the flags that were fetched later in the lifecycle / next session.
For fetchAndActivate - would we then handle an error as "nothing new to fetch" and wait for another opportunity to fetch freshest flags? In that case, I also agree that it can be a ProviderReady. My only concern (which might come from me not fully understanding Confidence/data science) is whether we report these events correctly.